test/functional/cursor_async_iterator.test.js

```diff
@@ -2,6 +2,7 @@
 
 const { expect } = require('chai');
 const Sinon = require('sinon');
+const { Promise: BluebirdPromise } = require('bluebird');
```

Comment:
I brought in bluebird because our current test just did `class P extends Promise` and I needed something that wouldn't extend the native promise.